---
title: "Data-mining biomarkers and perform testing"
author: "[Wilson Cai](www.stat.berkeley.edu/~wcai/), Nima Hejazi, & Alan Hubbard"
date: "`r Sys.Date()`"
output:
  # BiocStyle::html_document
vignette: >
  %\VignetteIndexEntry{Data-mining biomarkers and perform testing}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

## Introduction

The `adaptest` R package can be used to perform data-mining and high-dimensional statistical tests that is common in differential expression studies. The package utilizes a two stage procedure:
1. data-mining stage: reduce the dimension of biomarkers based on the associations of biomarkers with an exposure variable.
2. multiple testing stage: adjust for multiple testing to control false positives.
In this vignette, we illustrate how to use `adaptest` to perform such analysis, using a data set containing microarray expression measures.

---

## Data-mining stage

First, we load the `adaptest` package and the (included) `simpleArray` data set:

```{r setup_data}
# library(dplyr)
library(adaptest)
data(simpleArray)
"%ni%" = Negate("%in%")
```

In order to perform Targeted Minimum Loss-Based Estimation, we need three
separate data structures: (1) _W_, baseline covariates that could potentially
confound the association of biomarkers with the exposure of interest; (2) _A_,
the point exposure of interest; and (3) _Y_, the biomarkers of interest. All values in _A_ ought to be binarized, in order to avoid practical violations of
the assumption of positivity. To invoke the data-adaptive testing function (`adaptest`), we also need to specify the number of top biomarkers `n_top` to the data-mining algorithm, and the number of folds `n_fold` for cross-validation. The smaller `n_top` is, the more selective data-mining algorithm we have. The larger `n_fold` is, more folds are carried our in cross validaiton.

The TMLE-based biomarker discovery process can be invoked using the
`adaptest` function. The procedure is quite resource-intensive because it
evaluates the association of each individual potential biomarker (of which there
are 1e3 in the included data set) with an exposure of interest, while
accounting for potential confounding based on all other covariates included in
the design matrix. We demonstrate the necessary syntax for calling
`adaptest` below:

```{r biomarkerTMLE_eval, eval=FALSE}
adaptestout <- adaptest(Y = Y,
                       A = A,
                       W = NULL,
                       n_top = 10,
                       n_fold = 3)
```

```{r load_biomarkerTMLE_result, eval=FALSE}
data(adaptestOut)
```

The output of `adaptest` is an object of class `adaptest`, containing the following objects:
(1) top_index: (integer vector) - indices for the data-mining selected biomarkers
(2) top_colname: (character vector) - names for the data-mining selected biomarkers
(3) top_colname_significant_q: (character vector) - names for the data-mining selected biomarkers, which are significant after multiple testing stage
(4) DE: (numeric vector) - differential expression effect sizes for the biomarkers in \code{top_colname}
(5) p_value: (numeric vector) - p-values for the biomarkers in \code{top_colname}
(6) q_value: (numeric vector) - q-values for the biomarkers in \code{top_colname}
(7) significant_q: (integer vector) - indices of \code{top_colname} which is significant after multiple testing stage.
(8) mean_rank_top: (numeric vector) - average ranking across cross-validation folds for the biomarkers in \code{top_colname}
(9) folds: (origami::folds class) - cross validation object

After invoking `adaptest`, the resultant `adaptest` object will have the slots described above completely filled in. The statistical results of this procedure can be extracted using `summary` method.

---

## Visualization of Results

This package provides several plotting methods that can be used to visualize
the results of the TMLE-based biomarker discovery process.

The `plot` method for a `bioTMLE` object will produce a histogram of the
adjusted p-values of each biomarker (based on the Benjamini-Hochberg procedure
for controlling the False Discovery Rate) as generated by `limma::topTable`:

```{r pval_hist_limma_adjp}
# plot(x = limmaTMLEout, type = "pvals_adj")
```

Setting the argument `type = "pvals_raw"` will instead produce a histogram of
the raw p-values _(these are less informative and should, in general, not be
used for inferential purposes, as the computation producing these p-values
ignores the multiple testing nature of the biomarker discovery problem)_:

```{r pval_hist_limma_rawp}
# plot(x = limmaTMLEout, type = "pvals_raw")
```

Heatmaps are useful graphics for visualizing the relationship between measures
on genomic objects and covariates of interest. The `heatmap_ic` function
provides this graphic for `bioTMLE` objects, allowing for the relationship
between the exposure variable and some number of "top" biomarkers (as
determined by the call to `modtest_ic`) to be visualized. In general, the
heatmap for `bioTMLE` objects expresses how the contributions of each biomarker
to the Average Treatment Effect (ATE) vary across differences in the exposure
variable (that is, there is a causal interpretation to the findings). The plot
produced is a `ggplot2` object and can be modified in place if stored properly.
For our analysis:

```{r heatmap_limma_results}
# varInt_index <- which(names(colData(simpleArray)) %in% "benzene")
# designVar <- as.data.frame(colData(simpleArray))[, varInt_index]
# designVar <- as.numeric(designVar == max(designVar))

# heatmap_ic(x = limmaTMLEout, design = designVar, FDRcutoff = 0.05, top = 25)
```

The volcano plot is standard graphical tools for examining how changes in
expression relate to the raw p-value. The utility of such plots lies in their
providing a convenient way to identify and systematically ignore those genomic
objects that have extremely low p-values due to extremely low variance between
observations. The `volcano_ic` function provides much of the same
interpretation, except that the fold change values displayed in the x-axis refer
to changes in the _contributions of each biomarker to the Average Treatment
Effect_ (in standard practice, for microarray technology, these would be fold
changes in gene expression). The plot produced is a `ggplot2` object and can
be modified in place if stored properly. For our analysis:

```{r volcano_plot_limma_results}
# volcano_ic(adaptest = limmaTMLEout)
```

---

## Session Information

```{r sessionInfo, echo=FALSE}
sessionInfo()
```

