<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Data-Mining Biomarkers and High-Dimensional Testing • adaptest</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Data-Mining Biomarkers and High-Dimensional Testing">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">adaptest</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">1.2.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/differentialExpression.html">Data-Mining Biomarkers and High-Dimensional Testing</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/wilsoncai1992/adaptest">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Data-Mining Biomarkers and High-Dimensional Testing</h1>
                        <h4 class="author">Wilson Cai and <a href="https://nimahejazi.org">Nima Hejazi</a>
</h4>
            
            <h4 class="date">2018-10-30</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/wilsoncai1992/adaptest/blob/master/vignettes/differentialExpression.Rmd"><code>vignettes/differentialExpression.Rmd</code></a></small>
      <div class="hidden name"><code>differentialExpression.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>The <code>adaptest</code> R package can be used to perform data-mining and high-dimensional statistical tests that is common in differential expression studies. The package utilizes a two-stage procedure:</p>
<ol style="list-style-type: decimal">
<li><p>The <em>data-mining</em> stage: reduce the dimension of biomarkers based on the associations of biomarkers with an exposure variable.</p></li>
<li><p>The <em>multiple testing</em> stage: adjust for multiple testing to control false positives.</p></li>
</ol>
<p>In this vignette, we illustrate how to use <code>adaptest</code> to perform such analysis, using a data set containing microarray expression measures.</p>
<hr>
</div>
<div id="data-mining-stage" class="section level2">
<h2 class="hasAnchor">
<a href="#data-mining-stage" class="anchor"></a>Data-mining stage</h2>
<p>First, we load the <code>adaptest</code> package and the (included) <code>simpleArray</code> data set:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw">library</span>(adaptest)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="kw">data</span>(simpleArray)</a>
<a class="sourceLine" id="cb1-3" data-line-number="3"><span class="kw">set.seed</span>(<span class="dv">1234</span>)</a>
<a class="sourceLine" id="cb1-4" data-line-number="4">Y &lt;-<span class="st"> </span>simulated_array</a>
<a class="sourceLine" id="cb1-5" data-line-number="5">A &lt;-<span class="st"> </span>simulated_treatment</a></code></pre></div>
<p>In order to construct targeted minimum loss-based estimates for the purpose of hypothesis testing with data-adaptive parameters, we need three separate data structures:</p>
<ol style="list-style-type: decimal">
<li>
<em>W</em>, baseline covariates that could potentially confound the association of biomarkers with the exposure of interest;</li>
<li>
<em>A</em>, the point exposure of interest; and</li>
<li>
<em>Y</em>, the biomarkers of interest.</li>
</ol>
<p>All values in <em>A</em> ought to be binarized, in order to avoid practical violations of the assumption of positivity. To invoke the data-adaptive testing function (<code>adaptest</code>), we also need to specify the number of top biomarkers <code>n_top</code> to the data-mining algorithm, and the number of folds <code>n_fold</code> for cross- validation. The smaller <code>n_top</code> is, the more selective data-mining algorithm we have. The larger <code>n_fold</code> is, more folds are carried our in cross-validation.</p>
<p>The TMLE-based biomarker discovery process can be invoked using the <code>adaptest</code> function. The procedure is quite resource-intensive because it evaluates the association of each individual potential biomarker (of which there are <span class="math inline">\(1000\)</span> in the included data set) with an exposure of interest, while accounting for potential confounding based on all other covariates included in the design matrix. We demonstrate the necessary syntax for calling <code>adaptest</code> below:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1">adaptest_out &lt;-<span class="st"> </span><span class="kw"><a href="../reference/adaptest.html">adaptest</a></span>(<span class="dt">Y =</span> Y,</a>
<a class="sourceLine" id="cb2-2" data-line-number="2">                         <span class="dt">A =</span> A,</a>
<a class="sourceLine" id="cb2-3" data-line-number="3">                         <span class="dt">W =</span> <span class="ot">NULL</span>,</a>
<a class="sourceLine" id="cb2-4" data-line-number="4">                         <span class="dt">n_top =</span> <span class="dv">35</span>,</a>
<a class="sourceLine" id="cb2-5" data-line-number="5">                         <span class="dt">n_fold =</span> <span class="dv">5</span>,</a>
<a class="sourceLine" id="cb2-6" data-line-number="6">                         <span class="dt">learning_library =</span> <span class="kw">c</span>(<span class="st">"SL.glm"</span>, <span class="st">"SL.mean"</span>),</a>
<a class="sourceLine" id="cb2-7" data-line-number="7">                         <span class="dt">parameter_wrapper =</span> adaptest<span class="op">::</span>rank_DE,</a>
<a class="sourceLine" id="cb2-8" data-line-number="8">                         <span class="dt">absolute =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb2-9" data-line-number="9">                         <span class="dt">negative =</span> <span class="ot">FALSE</span>)</a></code></pre></div>
<p>The output of <code>adaptest</code> is an object of class <code>data_adapt</code>, containing the following major components:</p>
<ol style="list-style-type: decimal">
<li>
<code>top_index</code>: (integer vector) - indices for the biomarkers selected by way of data-mining.</li>
<li>
<code>top_colname</code>: (character vector) - names for the biomarkers selected by way of data-mining.</li>
<li>
<code>top_colname_significant_q</code>: (character vector) - names for the biomarkers selected by way of data-mining, which are significant after multiple testing stage.</li>
<li>
<code>DE</code>: (numeric vector) - differential expression effect sizes for the biomarkers in <code>top_colname</code>.</li>
<li>
<code>p_value</code>: (numeric vector) - p-values for the biomarkers in <code>top_colname</code>.</li>
<li>
<code>q_value</code>: (numeric vector) - q-values for the biomarkers in <code>top_colname</code>.</li>
<li>
<code>significant_q</code>: (integer vector) - indices of <code>top_colname</code> which is significant after multiple testing stage.</li>
<li>
<code>mean_rank_top</code>: (numeric vector) - average ranking across cross-validation folds for the biomarkers in <code>top_colname</code>.</li>
<li>
<code>folds</code>: (<code>origami::folds class</code>) - object of cross-validation folds.</li>
</ol>
<p>After invoking <code>adaptest</code>, the resultant <code>data_adapt</code> object will have the slots described above completely populated. Note that simply calling <code>objects()</code> on an object of class <code>data_adapt</code> will return more slots than those described above – many of these are auxiliary slots containing information that is likely not of interest to the user. The <span class="math inline">\(9\)</span> slots given above contain information that summarizes the findings of the data-adaptive hypothesis testing procedure. What’s more, even individually accessing these slots is inconvenient; thus, we provide a <code>summary</code> method to allow the statistical results of this procedure to be extracted easily. We demonstrate this below</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="kw">summary</span>(adaptest_out)</a></code></pre></div>
<pre><code>##    data-adaptive parameters Differential expression     p-values
## 1                         1             0.597999917 0.0036390849
## 2                         2             0.562077504 0.0053032638
## 3                         3             0.738512785 0.0003910451
## 4                         4             0.466686311 0.0133597241
## 5                         5             0.562575573 0.0067052878
## 6                         6             0.340545352 0.1354723778
## 7                         7             0.118216437 0.5293623032
## 8                         8             0.334149607 0.0523051808
## 9                         9            -0.295022663 0.1686641287
## 10                       10            -0.081931109 0.6598573297
## 11                       11             0.655897045 0.0023837050
## 12                       12            -0.122985292 0.5610206707
## 13                       13            -0.156552060 0.4029517418
## 14                       14            -0.065750874 0.7387750268
## 15                       15            -0.003383713 0.9875136952
## 16                       16            -0.041062298 0.8241596754
## 17                       17            -0.122255627 0.5210343651
## 18                       18            -0.250678811 0.2536461798
## 19                       19            -0.022472769 0.9123796877
## 20                       20            -0.022564500 0.9122673463
## 21                       21             0.125495321 0.5318335257
## 22                       22            -0.060315953 0.7246305162
## 23                       23             0.021052867 0.9132666459
## 24                       24             0.002261450 0.9909943652
## 25                       25             0.332692845 0.1089179334
## 26                       26            -0.298364287 0.1418775066
## 27                       27            -0.192006372 0.3552636248
## 28                       28             0.085360515 0.6723675842
## 29                       29             0.123067508 0.5145193129
## 30                       30            -0.381642301 0.0675600124
## 31                       31             0.088227679 0.6277388171
## 32                       32             0.354490171 0.0578489924
## 33                       33            -0.285088553 0.1093143193
## 34                       34            -0.229211821 0.1963214601
## 35                       35             0.210225084 0.3112861519
##      q-values
## 1  0.04245599
## 2  0.04640356
## 3  0.01368658
## 4  0.07793172
## 5  0.04693701
## 6  0.38197790
## 7  0.80931189
## 8  0.25308934
## 9  0.42166032
## 10 0.87158761
## 11 0.04171484
## 12 0.81815514
## 13 0.74227952
## 14 0.89162503
## 15 0.99099437
## 16 0.96151962
## 17 0.80931189
## 18 0.55485102
## 19 0.96861614
## 20 0.96861614
## 21 0.80931189
## 22 0.89162503
## 23 0.96861614
## 24 0.99099437
## 25 0.34781829
## 26 0.38197790
## 27 0.69079038
## 28 0.87158761
## 29 0.80931189
## 30 0.26273338
## 31 0.87158761
## 32 0.25308934
## 33 0.34781829
## 34 0.45808341
## 35 0.64088325</code></pre>
<p>In the table above, each column provides information about the results of the data-adaptive hypothesis testing procedure. In particular, columns 2-4 provide effect sizes, p-values, and q-values (after multiple testing correction) that may be of interest in scientifically interpreting the findings of the procedure.</p>
<p>As our goal in this vignette is to describe the properties of the <code>adaptest</code> software package and its operation, we omit a discussion of the statistical methodology implemented in this R package. A fully detailed technical account of the data-adaptive multiple testing procedure is available in <span class="citation">Cai, Hejazi, and Hubbard (n.d.)</span>, presently available on the arxiv. For an introduction to statistical inference procedures using data-adaptive target parameters, the interested reader is directed to <span class="citation">Hubbard, Kherad-Pajouh, and van der Laan (2016)</span>. For background on the Targeted Learning methodology, as well as recent advances, the canonical references are <span class="citation">van der Laan and Rose (2011)</span> and <span class="citation">van der Laan and Rose (2018)</span>.</p>
<hr>
</div>
<div id="interpret-and-visualize-results" class="section level2">
<h2 class="hasAnchor">
<a href="#interpret-and-visualize-results" class="anchor"></a>Interpret and Visualize Results</h2>
<p>This package provides several interpretation methods that can be used to tabular and visualize the results of the data-adaptive tests.</p>
<p>The <code>get_composition</code> method for a <code>adaptest</code> object will produce a table of composition of each data-adaptive parameters that is significant after multiple testing stage:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="kw"><a href="../reference/get_composition.html">get_composition</a></span>(<span class="dt">object =</span> adaptest_out, <span class="dt">type =</span> <span class="st">"small"</span>)</a></code></pre></div>
<pre><code>## [[1]]
##      3   5  10   4   9   1   6 987 257 519 649
## 1  0.2 0.2 0.6 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0
## 2  0.2 0.4 0.0 0.4 0.0 0.0 0.0 0.0 0.0 0.0 0.0
## 3  0.2 0.2 0.4 0.0 0.2 0.0 0.0 0.0 0.0 0.0 0.0
## 5  0.4 0.0 0.0 0.0 0.0 0.2 0.2 0.2 0.0 0.0 0.0
## 11 0.0 0.0 0.0 0.2 0.0 0.2 0.0 0.0 0.2 0.2 0.2
## 
## [[2]]
##      3   5  10   4   9   1   6 987 257 519 649   q-values
## 1  0.2 0.2 0.6 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.04245599
## 2  0.2 0.4 0.0 0.4 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.04640356
## 3  0.2 0.2 0.4 0.0 0.2 0.0 0.0 0.0 0.0 0.0 0.0 0.01368658
## 5  0.4 0.0 0.0 0.0 0.0 0.2 0.2 0.2 0.0 0.0 0.0 0.04693701
## 11 0.0 0.0 0.0 0.2 0.0 0.2 0.0 0.0 0.2 0.2 0.2 0.04171484</code></pre>
<p>Setting the argument <code>type = "big"</code> will instead produce a table of composition of each data-adaptive parameters before multiple testing stage, so that there are more columns. We omit running the code below since the output is large.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="co"># NOT RUN</span></a>
<a class="sourceLine" id="cb7-2" data-line-number="2"><span class="kw"><a href="../reference/get_composition.html">get_composition</a></span>(<span class="dt">object =</span> adaptest_out, <span class="dt">type =</span> <span class="st">"big"</span>)</a></code></pre></div>
<p>The <code>plot</code> method for a <code>adaptest</code> object will produce two plots that help user interpret the results. The first plot is a plot of sorted average CV-rank for all the biomarkers in the original data set (<code>Y</code>). The second plot is a plot of sorted q-values with labels corresponding to the indices of the data-adaptive parameter (as returned in <code>get_composition</code>)</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1"><span class="kw">plot</span>(adaptest_out)</a></code></pre></div>
<p><img src="differentialExpression_files/figure-html/plot-1.png" width="700"><img src="differentialExpression_files/figure-html/plot-2.png" width="700"></p>
<hr>
</div>
<div id="bioconductor-integration-with-summarizedexperiment" class="section level2">
<h2 class="hasAnchor">
<a href="#bioconductor-integration-with-summarizedexperiment" class="anchor"></a>Bioconductor Integration with <code>SummarizedExperiment</code>
</h2>
<p>Now, let’s try to acquire a taste for how we would use these algorithmic tools with objects common in computational biology – we’ll do this by performing the same analysis we did above, but using the core <code>SummarizedExperiment</code> container object and the popular <code>airway</code> data set as an example.</p>
<p>To start, let’s load the required packages:</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="kw">library</span>(SummarizedExperiment)</a>
<a class="sourceLine" id="cb9-2" data-line-number="2"><span class="kw">library</span>(airway)</a>
<a class="sourceLine" id="cb9-3" data-line-number="3"><span class="kw">data</span>(airway)</a></code></pre></div>
<p>For simplicity, we’ll restrict ourselves to just a random subset of the genes or transcripts available from the <em>airway</em> data set.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1">genes_sub &lt;-<span class="st"> </span><span class="kw">order</span>(<span class="kw">sample</span>(<span class="kw">seq_len</span>(<span class="dv">1000</span>)))</a>
<a class="sourceLine" id="cb10-2" data-line-number="2">air_reduced &lt;-<span class="st"> </span>airway[genes_sub, ]</a></code></pre></div>
<p>Generally, finding data-adaptive target parameters is a computationally and data-intensive procedure, requiring a fairly <strong>large sample size</strong>. To work with the relatively small <code>airway</code> data set, we’ll simply augment the data by <em>artificially</em> doubling it in size in a naive manner:</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1">simple_air &lt;-<span class="st"> </span><span class="kw">cbind</span>(air_reduced, air_reduced)</a></code></pre></div>
<p>We’ll now extract the variable of interest – a <code>factor</code> label for whether the unit received treatment or not – and coerce it to a binary <code>numeric</code> vector:</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1"><span class="co"># use a binary treatment variable (must be 0/1 only)</span></a>
<a class="sourceLine" id="cb12-2" data-line-number="2">dex_var =<span class="st"> </span><span class="kw">as.numeric</span>(<span class="kw">as.matrix</span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/SummarizedExperiment/topics/SummarizedExperiment-class">colData</a></span>(simple_air))[, <span class="dv">3</span>] <span class="op">-</span><span class="st"> </span><span class="dv">1</span>)</a></code></pre></div>
<p>Now, we can perform the same data-adaptive analysis we discussed above on this simplified version of the <em>airway</em> data set, simply by calling the <code>bioadaptest</code> wrapper function:</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1">airway_out &lt;-<span class="st"> </span><span class="kw"><a href="../reference/bioadaptest.html">bioadaptest</a></span>(<span class="dt">data_in =</span> simple_air,</a>
<a class="sourceLine" id="cb13-2" data-line-number="2">                          <span class="dt">var_int =</span> dex_var,</a>
<a class="sourceLine" id="cb13-3" data-line-number="3">                          <span class="dt">cntrl_set =</span> <span class="ot">NULL</span>,</a>
<a class="sourceLine" id="cb13-4" data-line-number="4">                          <span class="dt">n_top =</span> <span class="dv">5</span>,</a>
<a class="sourceLine" id="cb13-5" data-line-number="5">                          <span class="dt">n_fold =</span> <span class="dv">2</span>,</a>
<a class="sourceLine" id="cb13-6" data-line-number="6">                          <span class="dt">parameter_wrapper =</span> rank_DE)</a></code></pre></div>
<hr>
</div>
<div id="session-information" class="section level2">
<h2 class="hasAnchor">
<a href="#session-information" class="anchor"></a>Session Information</h2>
<pre><code>## R version 3.5.1 (2018-07-02)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 18.04.1 LTS
## 
## Matrix products: default
## BLAS: /usr/lib/x86_64-linux-gnu/openblas/libblas.so.3
## LAPACK: /usr/lib/x86_64-linux-gnu/libopenblasp-r0.2.20.so
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] airway_0.114.0              SummarizedExperiment_1.10.1
##  [3] DelayedArray_0.6.6          BiocParallel_1.14.2        
##  [5] matrixStats_0.54.0          Biobase_2.40.0             
##  [7] GenomicRanges_1.32.7        GenomeInfoDb_1.16.0        
##  [9] IRanges_2.14.12             S4Vectors_0.18.3           
## [11] BiocGenerics_0.26.0         nnls_1.4                   
## [13] adaptest_1.1.0              BiocStyle_2.8.2            
## 
## loaded via a namespace (and not attached):
##  [1] origami_1.0.0          xfun_0.4               listenv_0.7.0         
##  [4] lattice_0.20-35        colorspace_1.3-2       htmltools_0.3.6       
##  [7] yaml_2.2.0             rlang_0.3.0.1          pkgdown_1.1.0         
## [10] calibrate_1.7.2        GenomeInfoDbData_1.1.0 stringr_1.3.1         
## [13] zlibbioc_1.26.0        commonmark_1.6         tmle_1.3.0-1          
## [16] future_1.10.0          codetools_0.2-15       memoise_1.1.0         
## [19] evaluate_0.12          knitr_1.20             Rcpp_0.12.19          
## [22] backports_1.1.2        desc_1.2.0             XVector_0.20.0        
## [25] abind_1.4-5            fs_1.2.6               digest_0.6.18         
## [28] stringi_1.2.4          bookdown_0.7.21        rprojroot_1.3-2       
## [31] grid_3.5.1             tools_3.5.1            bitops_1.0-6          
## [34] magrittr_1.5           RCurl_1.95-4.11        crayon_1.3.4          
## [37] future.apply_1.0.1     MASS_7.3-51            Matrix_1.2-14         
## [40] data.table_1.11.9      xml2_1.2.0             SuperLearner_2.0-24   
## [43] assertthat_0.2.0       rmarkdown_1.10         roxygen2_6.1.0.9000   
## [46] R6_2.3.0               globals_0.12.4         compiler_3.5.1</code></pre>
<hr>
</div>
<div id="references" class="section level2 unnumbered">
<h2 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h2>
<div id="refs" class="references">
<div id="ref-cai2018data">
<p>Cai, Weixin, Nima S Hejazi, and Alan E Hubbard. n.d. “Data-Adaptive Statistics for Multiple Hypothesis Testing in High-Dimensional Settings.” <a href="https://arxiv.org/abs/1704.07008" class="uri">https://arxiv.org/abs/1704.07008</a>.</p>
</div>
<div id="ref-hubbard2016statistical">
<p>Hubbard, Alan E, Sara Kherad-Pajouh, and Mark J van der Laan. 2016. “Statistical Inference for Data Adaptive Target Parameters.” <em>The International Journal of Biostatistics</em> 12 (1): 3–19.</p>
</div>
<div id="ref-vdl2011targeted">
<p>van der Laan, Mark J, and Sherri Rose. 2011. <em>Targeted Learning: Causal Inference for Observational and Experimental Data</em>. Springer Science &amp; Business Media.</p>
</div>
<div id="ref-vdl2018targeted">
<p>———. 2018. <em>Targeted Learning in Data Science: Causal Inference for Complex Longitudinal Studies</em>. Springer Science &amp; Business Media.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#introduction">Introduction</a></li>
      <li><a href="#data-mining-stage">Data-mining stage</a></li>
      <li><a href="#interpret-and-visualize-results">Interpret and Visualize Results</a></li>
      <li><a href="#bioconductor-integration-with-summarizedexperiment">Bioconductor Integration with <code>SummarizedExperiment</code></a></li>
      <li><a href="#session-information">Session Information</a></li>
      <li><a href="#references">References</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Weixin Cai, Nima Hejazi.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
